tests/test_translator.py

```
@@ -0,0 +1,1198 @@
+
+    def test_regex_expand_with_literal_hash(self) -> None:
+        """Test expand() with literal # characters."""
+        regex = re.compile(r'^(\w+)$')
+        result = TranslatorByRegex.expand(regex, 'test', r'prefix#\1#suffix')
+        self.assertEqual(result, ['prefix#test#suffix'])
+
+    def test_regex_expand_with_dict_evaluation(self) -> None:
+        """Test expand() with dictionary expression evaluation."""
+        regex = re.compile(r'^(\w+)$')
+        result = TranslatorByRegex.expand(regex, 'test', r'prefix_{"a": "b"}["a"]_suffix')
+        self.assertEqual(result, ['prefix_b_suffix'])
```

Ruff (0.14.5)

Use a regular `assert` instead of unittest-style `assertEqual`

Replace `assertEqual(...)` with `assert ...`

(PT009)